lib/run-compilation.js

```diff
@@ -46,8 +41,15 @@ module.exports = function runCompilation(options, compilation, done) {
     });
 
   // eslint-disable-next-line no-unused-expressions
-  compilation.plugin && compilation.plugin('compilation', function (compilation) {
-    compilation.errors = compilation.errors.concat(errors);
-    compilation.warnings = compilation.warnings.concat(warnings);
+  compiler.plugin('compilation', function (compilation) {
```

Comment:
If removing the guard for `compilation.plugin` you can remove the eslint comment above.

I guess in more recent versions of webpack the race? condition for plugin being defined have been fixed? ie. Has this been verified to work consistently?